### ISSUE TITLE
[Regression] on_train_batch_begin callbacks with no batch number and size

### ISSUE THREAD

**Issue by KTTrev** [tensorflow/tensorflow#40703]
Yep this should work in TF2.1 as well, here's a full example (had to fix the metric code a bit):

```python
import tensorflow as tf

class Count(tf.keras.metrics.Metric):
  def __init__(self, name=None, dtype=None, **kwargs):
    super(Count, self).__init__(name, dtype, **kwargs)
    self.count = tf.Variable(0)

  def update_state(self, y_true, y_pred, sample_weight=None):
    first_tensor = tf.nest.flatten(y_true)[0]
    batch_size = tf.shape(first_tensor)[0]
    self.count.assign_add(batch_size)

  def result(self):
    return tf.identity(self.count)


class PrintInfo(tf.keras.callbacks.Callback):
  def on_train_batch_end(self, batch, logs):
    print('Batch number: {}'.format(batch))
    print('Samples seen this epoch: {}'.format(logs['counter']))

model = tf.keras.Sequential([tf.keras.layers.Dense(1)])
model.compile(optimizer='sgd', loss='mse', metrics=[Count(name='counter')])
x, y = tf.ones((10, 10)), tf.ones((10, 1))
model.fit(x, y, batch_size=2, callbacks=[PrintInfo()], verbose=2)
```

_Originally posted by @omalleyt12 in https://github.com/tensorflow/tensorflow/issues/39314#issuecomment-630325421_

I would like to use the same method, but with `on_train_batch_begin`, and it doesn't work. Actually the logs remain empty even after using the new metric. How can I use the batch size in a callback, with `on_train_batch_begin`?

**Comment by Saduf2019** [comment]
@KTTrev 
Please confirm if [this gist](https://colab.research.google.com/gist/Saduf2019/91c05885731c39ce65dbc80e2ae4416c/untitled237.ipynb) replicates the issue reported.

**Comment by KTTrev** [comment]
The gist is working well. The issue I'm facing is when I change `on_train_batch_end` with `on_train_batch_begin`.

**Comment by jvishnuvardhan** [comment]
@KTTrev Based on the [Keras docs](https://keras.io/guides/writing_your_own_callbacks/), I think the dictionary of `logs` is empty when you use `on_train_batch_begin`. On that page, there is an example also which list the `keys` available to access.

```
Start epoch 0 of training; got log keys: []
...Training: start of batch 0; got log keys: []
...Training: end of batch 0; got log keys: ['loss', 'mean_absolute_error']
...Training: start of batch 1; got log keys: []
...Training: end of batch 1; got log keys: ['loss', 'mean_absolute_error']
...Training: start of batch 2; got log keys: []
...Training: end of batch 2; got log keys: ['loss', 'mean_absolute_error']
...Training: start of batch 3; got log keys: []
...Training: end of batch 3; got log keys: ['loss', 'mean_absolute_error']
```

**Comment by KTTrev** [comment]
Exactly, but do you have any solution on how I can use the batch size within 'on_train_batch_begin' (which was suppose to be in the logs)?

**Comment by jvishnuvardhan** [comment]
@KTTrev Yes. This is regression issue. In TF2.1, you can do this 

```
  def on_train_batch_begin(self, batch, logs=None):
    keys = list(logs.keys()) # In TF2.2, this list is empty
    print("...Training: start of batch {}; got log keys: {}".format(batch, keys))
    print('Batch number: {}'.format(batch))
    print('Samples seen this epoch: {}'.format(logs['size']))
```

**Comment by omalleyt12** [comment]
@KTTrev Thanks for the issue!

`on_train_batch_begin` no longer receives any keys in the `logs`, could you explain your use case? I can advise on workarounds

**Comment by KTTrev** [comment]
![image](https://user-images.githubusercontent.com/47731576/85640784-02b0b300-b68d-11ea-8e37-75b0fd282a17.png)
Here is a screenshot of the model that I'm trying to build. You can notice that I'm using a function `MyFunction` at the output layer, which contain some parameters `param_1`, `param_2`, `param_3`, `param_4` . However these parameters are not static; each image going through the network uses diferent parameters. That's why, I want to update them during training, validation or prediction pahses, especially when using mini batches.

**Comment by saikumarchalla** [comment]
@KTTrev  As per the above comment, it is the expected behavior.  Closing this issue as of now. Please feel free to re open the issue if necessary. Thanks!.
